app.js

```diff
@@ -37,6 +56,26 @@ if (app.get("env") === "production") {
 app.use(require("body-parser").urlencoded({ extended: true }));
 app.use(session(sessionParms));
 
+/* ------------------ CSRF SETUP BELOW --------------------- */
+const csrf = require('host-csrf')
+
+app.use(cookieParser(process.env.SESSION_SECRET));
+app.use(express.urlencoded({ extended: false }));
+let csrf_development_mode = true;
+if (app.get("env") === "production") {
+    csrf_development_mode = false;
+    app.set("trust proxy", 1);
+}
+const csrf_options = {
+    protected_operations: ["PATCH"],
+    protected_content_types: ["application/json"],
+    development_mode: csrf_development_mode,
+};
+const csrf_middleware = csrf(csrf_options); //initialise and return middlware
```

Comment:
Great job! If you don’t want to add to every route you can also do app.use(csrf_middleware(req,res,next))